src/services/identity/identity.ts

```diff
@@ -0,0 +1,275 @@
+import { Veramo } from './agent.js';
+
+dotenv.config();
+
+export interface IIdentity {
+	agent?: VeramoAgent;
+
+	initAgent(): TAgent<any>;
+	createAgent?(agentId: string): Promise<VeramoAgent>;
+	createKey(type: 'Ed25519' | 'Secp256k1', agentId?: string): Promise<ManagedKeyInfo>;
+	getKey(kid: string, agentId?: string): Promise<ManagedKeyInfo>;
+	createDid(network: string, didDocument: DIDDocument, agentId?: string): Promise<IIdentifier>;
+	updateDid(didDocument: DIDDocument, agentId?: string): Promise<IIdentifier>;
+	deactivateDid(did: string, agentId?: string): Promise<boolean>;
+	listDids(agentId?: string): Promise<string[]>;
+	resolveDid(did: string, agentId?: string): Promise<DIDResolutionResult>;
+	resolve(didUrl: string): Promise<Response>;
+	getDid(did: string, agentId?: string): Promise<any>;
+	importDid(did: string, privateKeyHex: string, publicKeyHex: string, agentId?: string): Promise<IIdentifier>;
+	createResource(network: string, payload: ResourcePayload, agentId?: string): Promise<any>;
+	createCredential(
+		credential: CredentialPayload,
+		format: CredentialRequest['format'],
+		statusOptions: StatusOptions | null,
+		agentId?: string
+	): Promise<VerifiableCredential>;
+	verifyCredential(
+		credential: VerifiableCredential | string,
+		verificationOptions: VerificationOptions,
+		agentId?: string
+	): Promise<IVerifyResult>;
+	verifyPresentation(
+		presentation: VerifiablePresentation | string,
+		verificationOptions: VerificationOptions,
+		agentId?: string
+	): Promise<IVerifyResult>;
+	createStatusList2021(
+		did: string,
+		resourceOptions: ResourcePayload,
+		statusOptions: CreateStatusListOptions,
+		agentId: string
+	): Promise<CreateStatusList2021Result>;
+	updateStatusList2021(
+		did: string,
+		statusOptions: UpdateStatusListOptions,
+		publish?: boolean,
+		agentId?: string
+	): Promise<BulkRevocationResult | BulkSuspensionResult | BulkUnsuspensionResult>;
+	broadcastStatusList2021(
+		did: string,
+		resourceOptions: ResourcePayload,
+		statusOptions: BroadCastStatusListOptions,
+		agentId?: string
+	): Promise<boolean>;
+	checkStatusList2021(
+		did: string,
+		statusOptions: CheckStatusListOptions,
+		agentId?: string
+	): Promise<StatusCheckResult>;
+	searchStatusList2021(
+		did: string,
+		statusListName: string,
+		statusPurpose: 'revocation' | 'suspension',
+		agentId?: string
+	): Promise<any>;
+	revokeCredentials(
+		credential: VerifiableCredential | VerifiableCredential[],
+		publish: boolean,
+		agentId?: string
+	): Promise<RevocationResult | BulkRevocationResult>;
+	suspendCredentials(
+		credential: VerifiableCredential | VerifiableCredential[],
+		publish: boolean,
+		agentId?: string
+	): Promise<SuspensionResult | BulkSuspensionResult>;
+	reinstateCredentials(
+		credential: VerifiableCredential | VerifiableCredential[],
+		publish: boolean,
+		agentId?: string
+	): Promise<UnsuspensionResult | BulkUnsuspensionResult>;
+}
+
+export abstract class AbstractIdentity implements IIdentity {
```

Comment:
Let's switch naming convention here as well.
```suggestion
export abstract class AbstractIdentityService implements IIdentityService {
```